src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -470,17 +470,36 @@ private async Task HandleGCEvents(EventPipeEventSource source, int pid, Func<Tas
             _gcGraph.AllowReading();
         }
 
-        private void HandleProcessInfo(EventPipeEventSource source, Func<Task> stopFunc, CancellationToken token)
+        private async Task HandleProcessInfo(EventPipeEventSource source, Func<Task> stopFunc, CancellationToken token)
         {
-            source.Dynamic.AddCallbackForProviderEvent(MonitoringSourceConfiguration.EventPipeProviderName, "ProcessInfo", traceEvent =>
+            Action<TraceEvent, Action> processInfoHandler = (TraceEvent traceEvent, Action taskComplete) =>
             {
                 _processInfoCallback?.Invoke((string)traceEvent.PayloadByName("CommandLine"));
```

Comment:
I'm not sure this was covered in a previous review, but we should probably make this callback Func<Task> rather than just Action.

Reply:
The caller is non-async, so should I just wrap it in Task.Run?

Reply:
I updated it to be Func<string, Task> by capturing the command line local to the function and then invoking the callback after awaiting for the ProcessInfo event to be provided.